tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/clientserver/Checker.java

```diff
@@ -82,41 +82,40 @@ public Checker(int sleepMillis, TestData td, int fails) {
             while ((now > max) && !td.getMaxSimultaneousWorkers().compareAndSet(max, now)) {
                 max = td.getMaxSimultaneousWorkers().get();
             }
+
             if (fails > 0) {
                 Thread.sleep(millis / 2);
                 fails--;
                 RuntimeErrorException e = new RuntimeErrorException(new Error("fake error for Retry Testing"));
-                Utils.log(e.toString());
-
+                log(e.toString());
                 // We will countDown the latch in the finally block
-
                 throw e;
             }
 
-            Utils.log("Task " + taskId + " sleeping for " + millis + " milliseconds. " + now
-                    + " workers inside Bulkhead from " + td.getInstances() + " instances " + BAR.substring(0, now));
+            log("Task " + taskId + " sleeping for " + millis + " milliseconds. " + now
+                + " workers inside Bulkhead from " + td.getInstances() + " instances " + BAR.substring(0, now));
 
             Thread.sleep(millis);
 
-            Utils.log("Task " + taskId + " woke.");
+            if (td.getLatch() != null) {
+                td.getLatch().countDown();
+            }
```

Comment:
Why does this duplicate the `finally` block?

Reply:
good catch was a mistake thank you. done